AppDevAnnouncements/Classes/AnnouncementNetworking.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  AnnouncementNetworking.swift
+//  AppDevAnnouncements
+//
+//  Created by Gonzalo Gonzalez on 11/17/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import FutureNova
+
+public class AnnouncementNetworking {
+
+    static private var announcementPath: String?
+
+    static private var isConfigSetup: Bool = false
+
+    static private let networking: Networking = URLSession.shared.request
+    
+    static public func setupConfig(scheme: String, host: String, commonPath: String, announcementPath: String) {
+        if(scheme != "" && host != "" && commonPath != "" && announcementPath != "") {
+            Endpoint.setupEndpointConfig(scheme, host, commonPath)
+            self.announcementPath = announcementPath
+            self.isConfigSetup = true
+        } else {
+            self.isConfigSetup = false
+        }
+
+    }
+
+    static private func getAnnouncement() -> Future<Response> {
+        return networking(Endpoint.getAnnouncement(announcementPath!)).decode()
+    }
+
+    static internal func retrieveAnnouncement(completion: @escaping ((Announcement?) -> Void)) {
+        if isConfigSetup {
+            getAnnouncement().observe { result in
+                switch result {
+                case .value(let resp):
+                    let ref = resp.data[0]
+                    let announcement = Announcement(imageUrl: ref.imageUrl, subject: ref.subject, body: ref.body, ctaText: ref.ctaText, ctaAction: ref.ctaAction)
+                    completion(announcement)
+                case .error(let error):
+                    print(error.localizedDescription)
+                }
+            }
+        } else {
+            print("Did not setup config properly")
+        }
+    }
+
+}
+
+internal extension UIImageView {
+    func loadFromURL(photoUrl: String, completion: @escaping ((Bool) -> Void) = {_ in}) {
```

Comment:
I think there should be spaces next to the curly brackets. Also, instead of providing a default completion that does nothing, why not make it optional and `nil` by default? 

    func loadFromURL(photoUrl: String, completion: @escaping ((Bool) -> Void)? = nil) {

Then you change `completion(successful)` to `completion?(successful)`. I'm not sure which is better, just throwing out this alternative.

Reply:
FYI, making the completion optional gave me the following error:
`@escaping attribute only applies to function types`
so it seems functions cannot be optional.
Regardless, I'm thinking of just removing the default completion (I got this from another source) since we should probably add functionality for whether or not a picture was actually loaded or not (logging a console message or providing a default image).

Reply:
[This SO question](https://stackoverflow.com/questions/39618803/swift-3-optional-escaping-closure-parameter) offers an interesting explanation. Closures can definitely be optional, which apparently implicitly makes them escaping by default, so you would remove the @escaping attribute.

Reply:
Cool, I'll do that then for now and possibly make it a required field if necessary in the future.